### DISCUSSION TITLE
Location of the Dockerfiles for https://quay.io/repository/keycloak/keycloak

### DISCUSSION THREAD

**Discussion by tomasnorre** [keycloak/keycloak#16349] [Q&A] [answered]
Hi.

Anyone that can point me into the direction of the repository where the Dockerfiles for the images at https://quay.io/repository/keycloak/keycloak are stored? 

I have a Keycloak running with 19.0.3-legacy, which I want to update to 19.0.3 (not legacy) before I update to 20.0.2 (not legacy). 

When simply replacing the images and my command to following: 
```yaml
image: quay.io/keycloak/keycloak:19.0.3
command: start --http-enabled=true --optimized --spi-connections-jpa-legacy-migration-strategy=update --proxy edge --features-disabled=admin2
```

I get a routing error "/auth" path not found. So I cannot test if either the master REALM nor my custom REALM is working. 

I don't see any indications of problems in the logs. 

Anyone that can point me to the Dockerfiles, so that I can start doing some comparisons to learn a little more about the diff. I don't find the documentation helpful at this moment. Or I'm overlooking something important. 

Thanks in advance.

**Comment by ioemat** [comment, marked as answer]
Hi @tomasnorre,

the Dockerfile for the quarkus based implementation (non legacy) is within the keycloak project itself, see [quarkus/container/Dockerfile](https://github.com/keycloak/keycloak/tree/main/quarkus/container/Dockerfile).

For the legacy version the containers are described in a different project keycloak-containers, e.g. [in server/Dockerfile](https://github.com/keycloak/keycloak-containers/blob/main/server/Dockerfile).

Since the distributions work completely differently this probably doesn't help you much. I guess looking at https://www.keycloak.org/migration/migrating-to-quarkus will help more.

When you migrate from legacy to the quarkus distribution you probably need to pass `--http-relative-path=/auth` to have the same paths as before.

**Reply by tomasnorre** [reply]
Thanks. Will have a look. I don't know why I have overlooked this document. From reading it, the missing part is the `--http-relative-path=/auth` besides that it looks like I caught all changes.
